forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,447 @@
+        let lag = cur_duration_secs - ts;
+
+        let sync_status = if lag < 0 {
+            SyncStatus::Fast
+        } else if lag < EPOCH_DURATION_SECONDS * 3 / 2 {
+            // within 1.5 epochs
+            SyncStatus::Ok
+        } else if lag < EPOCH_DURATION_SECONDS * 5 {
+            // within 5 epochs
+            SyncStatus::Slow
+        } else {
+            SyncStatus::Behind
+        };
+
+        let base_fee = head.0.min_ticket_block().parent_base_fee().clone();
+
+        // blocks_per_tipset_last_finality = no of blocks till head / chain finality
+        let health = 100. * blocks_per_tipset_last_finality / BLOCKS_PER_EPOCH as f64;
+
+        Ok(Self {
+            lag,
+            health,
+            epoch: head.0.epoch(),
+            base_fee,
+            sync_status,
+            start_time,
+            network,
+            use_color,
+            default_wallet_address: None,
+            default_wallet_address_balance: None,
+        })
+    }
+
+    fn format(&self, now: DateTime<Utc>) -> anyhow::Result<String> {
+        let mut output = String::new();
+        if self.use_color {
+            writeln!(&mut output, "Network: {}", self.network())?;
+            writeln!(&mut output, "Uptime: {}", self.uptime(now))?;
+            writeln!(&mut output, "Chain: {}", self.chain_status())?;
+            writeln!(&mut output, "Chain health: {}", self.health())?;
+            writeln!(
+                &mut output,
+                "Default wallet address: {} {}",
+                self.wallet_address(),
+                self.wallet_balance()
+            )?;
+        } else {
+            writeln!(&mut output, "Network: {}", self.network().clear())?;
+            writeln!(&mut output, "Uptime: {}", self.uptime(now).clear())?;
+            writeln!(&mut output, "Chain: {}", self.chain_status().clear())?;
+            writeln!(&mut output, "Chain health: {}", self.health().clear())?;
+            writeln!(
+                &mut output,
+                "Default wallet address: {} {}",
+                self.wallet_address().clear(),
+                self.wallet_balance().clear()
+            )?;
+        }
+
+        Ok(output)
+    }
+}
+
+impl InfoCommand {
+    pub async fn run(&self, config: Config, opts: &CliOpts) -> anyhow::Result<()> {
+        let res = tokio::try_join!(
+            node_status((), &config.client.rpc_token),
+            chain_head(&config.client.rpc_token),
+            chain_get_name((), &config.client.rpc_token),
+            start_time(&config.client.rpc_token)
+        );
+        match res {
+            Ok((node_status, head, network, start_time)) => {
+                let cur_duration: Duration = SystemTime::now().duration_since(UNIX_EPOCH)?;
+                let blocks_per_tipset_last_finality =
+                    node_status.chain_status.blocks_per_tipset_last_finality;
+
+                let mut node_status_info = NodeStatusInfo::new(
```

Comment:
This doesn't need to be `mut` if you refactor it slightly.

Reply:
I get https://github.com/ChainSafe/forest/actions/runs/5256500061/jobs/9497916282?pr=2578#step:6:1508 when moving wallet and balance fields as params

Reply:
refactored to remove `use_color` field